mimic/model/behaviors.py

```diff
@@ -312,3 +323,72 @@ def delete_behavior(kl_self, request, behavior_id):
                 delete_behavior)
 
     return BehaviorAPI
+
+
+sequence_docstring = """
+    Sometimes a sequence of behaviors occur when you try to trigger an
+    event in a predictable pattern.
+
+    Takes one parameter, ``behaviors``, which is a list of specifications
+    of other behaviors, similar to those specified in the request to
+    create a behavior, with the addition of a behavior with a name of
+    "default" that means default success.
+
+    Each time the criterion for this behavior is matched, the next
+    behavior is executed, looping back to the beginning when the list of
+    behaviors is exhausted.  In other words, this creation behavior is
+    stateful.
+
+    Note that the behavior specifications here do not need a criterion,
+    since the criterion is specified for the behavior overall, and each
+    behavior is unconditionally executed in sequence.
+
+    For example, to specify an alternating sequence of success and then
+    failure when the criterion for the ``sequence`` behavior is matched::
+
+        {
+            "behaviors": [
+                {
+                    "name": "default"
+                },
+                {
+                    "name": "fail",
+                    "parameters": {
+                        "code": 500,
+                        "message": "synthetic error"
+                    }
+                }
+            ]
+        }
+"""
+
+
+def _sequence_behavior(event):
+    """
+    A convenience function that should for :class:`EventDescription` that,
+    given an event, produces a generic behavior-creator that provides this
+    sequence behavior, which is named "sequence".
+
+    :param event: an instance of :class:`EventDescription`
+    :return: a callable behavior-creator as described above
+    """
+    @event.declare_behavior_creator("sequence")
+    def sequence(parameters):
+        behavior_specification = parameters["behaviors"]
+        behavior_objects = cycle([
+            (
+                event.create_behavior(behavior["name"],
+                                      behavior["parameters"])
+                if behavior["name"] != "default"
+                else event.default_behavior
+            )
+            for behavior in behavior_specification
+        ])
+
+        def rotating_behavior(*args, **kwargs):
+            current = next(behavior_objects)
+            return current(*args, **kwargs)
+        return rotating_behavior
+
+    sequence.__doc__ = sequence_docstring
```

Comment:
The rule, by the way, is that Python will assign a docstring if the first statement in the suite (class or function body) is a string literal.  `"foo".format("bar")` is not a literal, it's a complex expression, so it doesn't count.  But this would.